### PR TITLE
Add KlusterletConfig.

### PR DESCRIPTION
**Pull request by xuezhaojun**
https://issues.redhat.com/browse/ACM-2252

### CODE REVIEW

File: klusterletconfig/v1alpha1/types.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+	// +optional
+	Spec KlusterletConfigSpec `json:"spec,omitempty"`
+
+	// Status defines the observed state of KlusterletConfig
+	// +optional
+	Status KlusterletConfigStatus `json:"status,omitempty"`
+}
+
+// KlusterletConfigSpec defines the desired state of KlusterletConfig, usually provided by the user.
+type KlusterletConfigSpec struct {
```
Comment (zhiweiyin318) [inline klusterletconfig/v1alpha1/types.go:28]: @elgnay do we define the proxy fileds in this PR too?

Comment (elgnay) [reply klusterletconfig/v1alpha1/types.go:28]: It will be in a separated PR.

File: klusterletconfig/v1alpha1/types.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:path=klusterletconfigs
```
Comment (qiujian16) [inline klusterletconfig/v1alpha1/types.go:11]: is this cluster scoped?

File: klusterletconfig/v1alpha1/types.go [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+	// PullSecret is the name of image pull secret.
+	// +optional
+	PullSecret corev1.ObjectReference `json:"pullSecret,omitempty"`
+
+	// NodePlacement enables explicit control over the scheduling of the agent components.
+	// If the placement is nil, the placement is not specified, it will be omitted.
+	// If the placement is an empty object, the placement will match all nodes and tolerate nothing.
+	// +optional
+	NodePlacement *NodePlacement `json:"nodePlacement,omitempty"`
```
Comment (qiujian16) [inline klusterletconfig/v1alpha1/types.go:41]: I think we already have nodePlacement defined in ocm api. Do you want to reuse it?

File: klusterletconfig/v1alpha1/types.go (hunk 2 above)
Comment (xuezhaojun) [reply klusterletconfig/v1alpha1/types.go:11]: `// +kubebuilder:resource:scope=Cluster` added.

File: klusterletconfig/v1alpha1/types.go (hunk 3 above)
Comment (xuezhaojun) [reply klusterletconfig/v1alpha1/types.go]: Agree, fix with reused NodePlacement.

**Pull request merged**